modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM574ChangeTheStatusOfAnAPIToPrototypedThroughThePublisherRestAPITestCase.java

```diff
@@ -109,132 +100,84 @@ public void testCreateAnAPIThroughThePublisherRest() throws Exception {
-        assertTrue(creationResponse.getData().contains("PUBLISHED"),
-                apiNameTest + "  status not updated as Prototyped");
-
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, APILifeCycleAction.PUBLISH.getAction());
+        assertTrue(restAPIPublisher.getLifecycleStatus(apiId).getData().equals("Published"), apiNameTest +
+                "status not updated as Published");
         //Check whether published API is available in publisher
-        HttpResponse prototypedApiResponse = (apiPublisher.getAPI
-                (apiNameTest, apiProvider, apiVersion));
-        JSONObject prototypedApiObject = new JSONObject(prototypedApiResponse.getData());
-        JSONObject allApiObject = new JSONObject(prototypedApiObject.getString("api"));
-        assertFalse(prototypedApiObject.getBoolean("error"), apiNameTest +
+        HttpResponse publishedApiResponse = restAPIPublisher.getAPI(apiId);
+        assertEquals(publishedApiResponse.getResponseCode(), Response.Status.OK.getStatusCode(), apiNameTest +
                 " is not visible in publisher");
-        assertTrue(allApiObject.getString("name").equals(apiNameTest),
-                apiNameTest + " is not visible in publisher");
-        assertTrue(allApiObject.getString("status").equals("PUBLISHED"),
-                "Status of the " + apiNameTest + "is not a valid status");
-
+        assertTrue(publishedApiResponse.getData().contains(apiNameTest), apiNameTest + " is not visible in " +
+                "publisher");
     }
 
     @Test(groups = {"wso2.am"}, description = "Change the status of the API to DEPRECATED through" +
             " the publisher rest API ", dependsOnMethods = "testChangeTheStatusOfTheAPIToPublished")
     public void testChangeTheStatusOfTheAPIToDeprecated() throws Exception {
 
         //Change the status PUBLISHED to DEPRECATED
-        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest
-                (apiNameTest, apiProvider, APILifeCycleState.DEPRECATED);
-        updateRequest.setRequireResubscription("true");
-        HttpResponse creationResponse = apiPublisher.changeAPILifeCycleStatus(updateRequest);
-        assertTrue(creationResponse.getData().contains("DEPRECATED"),
-                apiNameTest + "  status not updated as Prototyped");
-
-        //Check whether deprecated API is available in publisher
-        HttpResponse prototypedApiResponse = (apiPublisher.getAPI
-                (apiNameTest, apiProvider, apiVersion));
-        JSONObject prototypedApiObject = new JSONObject(prototypedApiResponse.getData());
-        JSONObject allApiObject = new JSONObject(prototypedApiObject.getString("api"));
-        assertFalse(prototypedApiObject.getBoolean("error"), apiNameTest +
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, APILifeCycleAction.DEPRECATE.getAction());
+        assertTrue(restAPIPublisher.getLifecycleStatus(apiId).getData().equals("Deprecated"), apiNameTest +
```

Comment:
Shall we use the constant - `APILifeCycleState.PROTOTYPED.getState()` here?

Reply:
And switch the equals method to avoid NPE

Reply:
fixed